File: 01/p2.py
import sys

def cycle_counter(pointer: int, distance: int, range: int) -> int:
    cycles = 0
    new_pointer = pointer + distance 

    while(new_pointer < 0):
        new_pointer += 100
        cycles += 1

    while(new_pointer >= range):
        new_pointer -= 100
        cycles += 1

    return cycles

dial_pointer = 50;
dial_range = 100;
counter = 0;

for line in sys.stdin:
    line = line.rstrip("\n")
    dir = line[0]
    dist = int(line[1:])

    if dir == 'L':
        dir = -1
    else:
        dir = 1
    
    counter += cycle_counter(dial_pointer, dir * dist, dial_range)
    dial_pointer = (dial_pointer + (dir * dist)) % dial_range
    
print(counter)

File: 01/p2.cpp
#include <iostream>
#include <sstream>
#include <string>

// % cpp doesn't handle negatives
// like you expect python would lol
int modulo(int a, int b) { return (a % b + b) % b; }

int main(int argc, char *argv[]) {

  int dial_pointer = 50;
  int dial_range = 100;
  int counter = 0;
  std::string line;

  while (std::getline(std::cin, line)) {
    std::istringstream iss(line);

    char c;
    int dist = 0;

    iss >> c >> dist;

    // go backwards if L, forwards if R
    int dir = (c == 'L') ? -1 : 1;

    dial_pointer = modulo(dial_pointer + (dir * dist), dial_range);

    if (dial_pointer == 0) {
      counter++;
    }
  }

  std::cout << counter << std::endl;

  return 0;
}
